CONTRIBUTING.md

```diff
@@ -0,0 +1,136 @@
+Since proposals begin (and will often end) with the filing of an issue, even
+small changes can go through the proposal process if appropriate.
+Deciding what is appropriate is matter of judgment we will refine through
+experience.
+If in doubt, file a proposal.
+
+### Process
+
+- Create an issue describing the proposal.
+
+- Like any GitHub issue, a Proposal issue is followed by an initial discussion
+  about the suggestion. For Proposal issues:
+  - The goal of the initial discussion is to reach agreement on the next step:
+    (1) accept, (2) decline, or (3) ask for a design doc.
+  - The discussion is expected to be resolved in a timely manner.
+  - If the author wants to write a design doc, then they can write one.
+  - A lack of agreement means the author should write a design doc.
+  - If there is disagreement about whether there is agreement,
+    the project's main commiter is the arbiter.
+
+- If a Proposal issue leads to a design doc:
+  - The design doc should be checked in to `design/NNNN-shortname.md`,
+    where `NNNN` is the GitHub issue number and `shortname` is a short name
+    (a few dash-separated words at most).
+  - The design doc should follow the template found below.
+  - The design doc should address any specific issues asked for during the
+    initial discussion.
+  - It is expected that the design doc may go through multiple checked-in revisions.
+  - New design doc authors may be paired with a design doc "shepherd" to help work
+    on the doc.
+  - Comments by others can be made on the GitHub issue.
+
+- Once comments and revisions on the design doc wind down, there is a final
+  discussion about the proposal.
+  - The goal of the final discussion is to reach agreement on the next step:
+    (1) accept or (2) decline.
+  - The discussion is expected to be resolved in a timely manner.
+  - A lack of agreement means decline.
+  - If there is disagreement about whether there is agreement, the
+    project's main commiter is the arbiter.
+
+- The author (and/or other contributors) do the work as described by the
+  "Implementation" section of the proposal.
+
+### Template
+```markdown
+# Proposal: [Title]
+
+Author(s): [Author Name, Co-Author Name]
+
+Last updated: [Date]
+
+## Abstract
+
+[A short summary of the proposal.]
+
+## Background
+
+[An introduction of the necessary background and the problem being solved by the proposed change.]
+
+## Proposal
+
+[A precise statement of the proposed change.]
+
+## Rationale
+
+[A discussion of alternate approaches and the trade offs, advantages, and disadvantages of the specified approach.]
+
+## Implementation
+
+[A description of the steps in the implementation, who will do them, and when.]
+
+## Open issues (if applicable)
+
+[A discussion of issues relating to this proposal for which the author does not
+know the solution. This section may be omitted if there are none.]
+```
+
+## Implementation
+
+Every contribution must conform to the following rules:
```

Comment:
Right, I guess I was wondering if there was a need to be a little more explicit.